src/Awaitable.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Interop\Async;
+
+/**
+ * Simple interface for awaitable objects representing the future value of asynchronous operations.
+ */
+interface Awaitable
+{
+    /**
+     * Registers a callback to be invoked when the awaitable is resolved.
+     *
+     * @param callable(\Throwable|\Exception $exception = null, mixed $result = null) $onResolved
```

Comment:
Should probably be `callable(\Throwable|\Exception|null $exception, mixed $result) $onResolved`